### ISSUE TITLE
Fieldname naming convention

### ISSUE THREAD

**Issue by lindahua** [JuliaStats/Distributions.jl#395]
There have been some debates about naming conventions of fieldnames (*e.g.* #340). Actually, looking at the codes, we can find inconsistencies in naming. 

I am opening this issue, for the purpose of establishing a naming convention that we should follow to name parameter fields in distribution types. 

Below is a tentative proposal:
```julia
α   # \alpha: shape parameters (e.g. Gamma)
β   # \beta: scale parameters (e.g. Gamma, Exponential)
μ   # \mu: mean parameter (e.g. Normal, Poisson)
ν   # \nu: degree of freedom (e.g. Chisq)
σ   # \sigma: standard deviation (e.g. Normal)
λ   # \lambda: non-centrality
p   # probability, success-rate (e.g. Bernoulli, Binomial, Geometric)
n   # the number of trials (e.g. Binomial, Multinomial)
```
For other cases, the general guideline is to follow what is being used in Wikipedia.

Any comments? I may try to go through the distributions and implement this rule (possibly this week or next), and also put the naming convention to the document.

---

EDIT: Now a modification is adopted, according to @simonbyrne's suggestion:

```julia
μ, σ: # location and scale parameter in location-scale families (e.g. Normal, LogNormal, etc)
θ:  # \theta: scale parameter, in scale-only families (e.g. Exponential, and Gamma)
```

Also, according to the current convention, there's another rule:

```julia
(a, b):  # indicate two bounds of a bounded support (e.g. Arcsine, Uniform, Triangular)
```

**Comment by andreasnoack** [comment]
It would be great eith consistency in the field names so thanks for opening this. I support your suggested naming convention. Should we also decide on a localtion/non-centrality parameter name?

**Comment by lindahua** [comment]
I think non-centrality may use `ν (\nu)`, as in Wikipedia (Noncentral T). 

For location, if that parameter represents the mean (e.g. Normal), we should just use `mu`. For others, we may refer to Wikipedia. I will bring up the issue if I come across names that I am not sure of.

**Comment by lindahua** [comment]
We can use `loc` for location parameter, if there's no conventional name for that specific parameter in certain distributions.

**Comment by andreasnoack** [comment]
On Wikipedia, It appears that the non-central beta, chi-square and F distributions use λ. I think I've also seen ν used for the degrees of freedom in some places, so maybe we should use λ for location/non-centrality.

**Comment by lindahua** [comment]
Sounds good, then let's use `λ` for noncentrality. 

In terms of location (e.g. the one in `TriWeight`), not sure `λ` is a proper choice.

**Comment by simonbyrne** [comment]
My 2 cents:
* As I mentioned in #340, `β` is more commonly used as the inverse-scale, [e.g. here](https://books.google.co.uk/books?id=TNYhnkXQSjAC&lpg=PP1&dq=bayesian%20data%20analysis&pg=PA586#v=onepage&q&f=false). Could we use `θ` for the scale (this seems quite common).
* `ν` (`\nu`) is often used as degrees-of-freedom.
* `λ` is typically used as the rate of a Poisson distribution. We could also use the inverse-scale parameter of the Exponential, due to them both arising in a Poisson process.
* `μ`,`σ` are quite often used as parameters in location-scale families, though we could also use `μ`,`θ`.

**Comment by lindahua** [comment]
Seems that the only thing that we have not reached consensus is what we should use for scale (except for `Normal`, where `σ` is widely used for representing standard dev).

I will do a review of what Wikipedia uses in various distributions.

**Comment by simonbyrne** [comment]
My preference would be:
* `μ`,`σ` for location-scale families
* `θ` for scale

**Comment by nalimilan** [comment]
@simonbyrne Do you have other examples where `θ` is used for the scale? It doesn't look like there's a way to be consistent with all uses, but at least we choose try to retain the most common definition.

I've seen `β` used both for the shape and the inverse scale, as in various parameterization of the [log-logistic distribution] (https://en.wikipedia.org/wiki/Log-logistic_distribution) (where `α` is sometimes the scale...). This inconsistency might indeed be in favour of using something else, like `θ`.

**Comment by lindahua** [comment]
I did a simple survey, over Wikipedia, and get the following:

Normal: σ for scale (i.e. standard dev)
Gamma: θ for scale
Exponential: λ for rate, β for scale
Cauchy: γ for scale
LogNormal: s for scale
Laplace: b for scale
(General) location-scale family: μ for location, σ for scale

Show each individual distribution I looked at, it uses a different symbol for representing the scale parameter. I am now leaning towards @simonbyrne's suggestion:

- (μ, σ) for location-scale families
- θ for scale-only.

**Comment by StefanKarpinski** [comment]
I'm barely qualified to have an opinion here, but to me `μ, σ` suggest mean and standard deviation, which isn't always what location and scale parameters mean, right?

**Comment by lindahua** [comment]
@StefanKarpinski See https://en.m.wikipedia.org/wiki/Location-scale_family

`mu` and `sigma` are used for general location-scale family as location and scale parameters.

**Comment by StefanKarpinski** [comment]
Ok, carry on! Glad to see this happening.

**Comment by lindahua** [comment]
Implemented by #400
